lib/plugins/wallet/cardano/api.js

```diff
@@ -0,0 +1,161 @@
+const BN = require('../../../bn')
+
+const unitScale = 6
+
+class CardanoAPI {
+  constructor(client, port, hostname = 'localhost') {
+    this.client = client
+    this.port = port
+    this.hostname = hostname
+  }
+
+  async fetch(endpoint, params, data, appendData = []) {
+    const method = data ? 'POST' : 'GET'
+
+    const url = `https://${this.hostname}:${this.port}/api/v1/${endpoint}`
+
+    return Promise.resolve(true)
+      .then(() => {
+        return this.client({
+          method,
+          url,
+          params,
+          data
+        })
+      })
+      .then(response => {
+        if (response.error) throw r.error
+
+        const responseData = Array.isArray(response.data.data)
+          ? [...response.data.data, ...appendData]
+          : response.data.data
+
+        if (
+          method === 'GET' &&
+          response.data.meta.pagination.totalPages >
+            response.data.meta.pagination.page
+        ) {
+          return this.fetch(
+            endpoint,
+            {
+              ...params,
+              page: (params.page || 1) + 1
+            },
+            data,
+            responseData
+          )
+        }
+
+        return responseData
+      })
+  }
+
+  async walletBalance(walletId, index, confirmations = 1) {
+    return this.fetch('transactions', {
+      walletId,
+      accountIndex: index
+    })
+      .then(transactions =>
+        transactions.reduce((acc, item) => {
+          if (item.confirmations < confirmations) {
+            return acc
+          }
+
+          return acc + (item.direction === 'outgoing' ? -1 : 1) * item.amount
```

Comment:
I would rather use mapping
```
sign = { incoming: +1, outgoing: -1}[item.direction]
```
so that it will crash if it doesn't parse direction properly.

Reply:
:+1: